tests/unit/payment-method/test-class-giropay.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Class Sepa_Test
+ * @package WCPay\Payment_Gateway\Tests
+ */
+
+namespace WCPay\Payment_Method;
+
+use DateTime;
+use PHPUnit_Framework_MockObject_MockObject;
+use WC_Payment_Gateway_WCPay_Subscriptions_Compat;
+use WC_Payments_Account;
+use WC_Payments_Action_Scheduler_Service;
+use WC_Payments_API_Client;
+use WC_Payments_API_Intention;
+use WC_Payments_Customer_Service;
+use WC_Payments_Token_Service;
+use WC_Payments_Utils;
+use WCPay\Constants\Payment_Capture_Type;
+use WCPay\Constants\Payment_Initiated_By;
+use WCPay\Constants\Payment_Type;
+use WCPay\Payment_Information;
+use WP_UnitTestCase;
+use WP_User;
+
+/**
+ * WCPay\Payment_Gateway\Sepa Unit tests
+ */
+class Giropay_Test extends WP_UnitTestCase {
```

Comment:
Maybe in the future we can optimize these tests so they extend a base class and all these setup code will not need to be duplicated.

Reply:
I may attempt to do this when I shortly implement Sofort as a payment method to avoid duplicating code over and over.